recipes/dd-agent.rb

```diff
@@ -39,6 +39,11 @@
 # Set the correct Agent startup action
 agent_start = node['datadog']['agent_start'] ? :start : :stop
 
+# use either the site option or the url one, the latter having priority.
+node.run_state['dd_url'] = 'https://app.datadoghq.com'
+node.run_state['dd_url'] = 'https://app.' + node['datadog']['site'] unless node['datadog']['site'].nil?
+node.run_state['dd_url'] = node['datadog']['url'] unless node['datadog']['url'].nil?
```

Comment:
I'd avoid using the run state (to me it's a code smell, even though I don't have chef code style guidelines to justify this). This logic could live in the function `template_vars` below instead